osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/application/services/SecretManagementService.java

```diff
@@ -70,10 +70,20 @@ public SecretManagementService(RsaEncryptionProvider rsaEncryptionProvider,
     @Override
     public byte[] getDlmsAuthenticationKey(String deviceIdentification) {
 
+        GetSecretsResponse response;
+
         try {
             GetSecretsRequest request = getSoapRequestForKey(deviceIdentification,
-                    SecretType.E_METER_AUTHENTICATION_KEY);
-            GetSecretsResponse response = secretManagementClient.getSecretsRequest(request);
+                SecretType.E_METER_AUTHENTICATION_KEY);
+
+            response = secretManagementClient.getSecretsRequest(request);
+        }
+        catch (Exception e) {
+            throw new ConnectionException("Error while communicating with secret management "
```

Comment:
This is a very general message. If you have a very distinct set of possible error scenarios, consider creating specific catches/exception messages for them.

Reply:
We do not know what can go wrong currently in production! This is a measure to avoid crashing.

Reply:
I agree, when a malformed message is rejected, the connection error will be thrown. This should be improved.